source/grid/grid_tools.cc

```diff
@@ -2553,7 +2553,7 @@ namespace GridTools
     if (matrix.m() == spacedim)
       for (int i = 0; i < spacedim; ++i)
         for (int j = 0; j < spacedim; ++j)
-          distance(i) = matrix(i,j) * point1(j);
+          distance(i) += matrix(i,j) * point1(j);
```

Comment:
Do I understand you right that the code before was simply wrong, but that nobody noticed because we happened to only run testcases where it didn't matter?